packages/workflow/src/internals.ts

```diff
@@ -577,6 +686,29 @@ export class Activator implements ActivationHandler {
     }
   }
 
+  public dispatchBufferedUpdates(): void {
+    const bufferedUpdates = this.bufferedUpdates;
+    while (bufferedUpdates.length) {
+      const foundIndex = bufferedUpdates.findIndex((update) => update.name && this.updateHandlers.has(update.name));
```

Comment:
nit: `(update) => this.updateHandlers.has(update.name as string)` or `(update) => this.updateHandlers.has(update.name ?? '')` rather than `(update) => update.name && this.updateHandlers.has(update.name)`

Reply:
Why do you want to look up and use the value associated with empty string? I don't believe empty string is ever a valid hjandler key, so that looks wrong / potentially buggy to me, and my code deliberately does it differently than signal, but I was going to ask about why Signal does it that way.

Reply:
Discussed offline: the types are failing to reflect that `update.name` is guaranteed to be a string. I've switched both the update and signal versions to use a type assertion.